### DISCUSSION TITLE
Finite strain simulations with the Maxwell Viscoelastic model

### DISCUSSION THREAD

**Discussion by veevee1208** [idaholab/moose#16600] [Q&A Modules: Solid mechanics] [answered]
Hello MOOSE team,

I have been running some finite strain simulations with the viscoelastic Kelvin Voigt model and I do not have any issues with convergence. However, when I try to run the finite strain simulation with the Maxwell model, I seem to run into a convergence issue . I tried using the PJFNK and AD but they seem to be of no effect. I was wondering if someone has experience running viscoelastic models on MOOSE and have any suggestions ? I have attached a copy of my input file.

Any help will be appreciated.

Best Regards
Vishal
[Maxwell-finite.txt](https://github.com/idaholab/moose/files/5755614/Maxwell-finite.txt)

**Comment by crswong888** [comment, marked as answer]
Hi @veevee1208, I tried playing around with your input file and I improved a bit the performance, but I think the problem you are having is related to very high strains. Please make sure that all of your units are consistent and that the forces and boundary conditions you're applying are realistic. I changed the axial force in the x-direction from 10e3 to just 10, and at 30 seconds the strain in the x-direction is 380%, which is a really high number for most solid materials (even viscoelastic ones) and leads to poor conditioning in the Jacobian.

You might find my version of your input file helpful: [github.com/crswong888/scorpion/blob/master/inputs/users/Maxwell-finite.i](github.com/crswong888/scorpion/blob/master/inputs/users/Maxwell-finite.i). It diverges at about 30 seconds though.

If you continue to have issues, it may be helpful for me to take a look at the version of your input file for which you had none.
